test/juxt/site/graphql_test.clj

```diff
@@ -301,6 +301,164 @@ type Mutation {
 (deftest mutation-test
   (mutation))
 
+(defn mutation-with-validation-directive []
+  (let [schema "
+schema { query: Query mutation: Mutation }
+type Query { person: Person }
+type Person { id: ID @site(a: \"xt/id\") name: String }
+input PersonInput { name: String! id: ID }
+
+type Mutation {
+  addPerson(
+    person: PersonInput
+  ): Person @site(
+    validation: {
+      person: \"[:map [:id {:optional true} :any] [:name [:string {:min 1 :max 20}]]]\"
+    }
+  )
+}"
+
+        query "
+mutation { addPerson(
+  person: {id: \"https://example.org/persons/mal\" name: \"Malcolm Sparks\"}) { id name }}
+"]
+
+    (submit-and-await!
+     [[:xtdb.api/put access-all-areas]
+
+      [:xtdb.api/put
+       {:xt/id "https://example.org/graphql"
+        :doc "A GraphQL endpoint"
+        :juxt.http.alpha/methods #{:post :put :options}
+        :juxt.http.alpha/acceptable "application/graphql"
+        :juxt.site.alpha/put-fn 'juxt.site.alpha.graphql/put-handler
+        :juxt.site.alpha/post-fn 'juxt.site.alpha.graphql/post-handler}]])
+
+    ;; Install a GraphQL schema at /graphql
+    (let [response (*handler*
```

Comment:
we could do with abstracting these out into 'put-schema' and 'run-mutation' etc functions, though probably leave it for now unless its easy